src/main/java/uk/gov/hmcts/darts/authorisation/configuration/AuthorisationAspect.java

```
@@ -0,0 +1,259 @@
+
+    private static final Pattern HEARINGS_ID_PATH_PATTERN = Pattern.compile(
+        String.format(PATH_PARAM_REGEX, HEARINGS_PATH));
+
+    private static final Pattern AUDIO_REQUESTS_ID_PATH_PATTERN = Pattern.compile(
+        String.format(PATH_PARAM_REGEX, AUDIO_REQUESTS_PATH));
+
+    private static final Pattern AUDIOS_ID_PATH_PATTERN = Pattern.compile(
+        String.format(PATH_PARAM_REGEX, AUDIOS_PATH));
+
+    private static final Pattern TRANSCRIPTIONS_ID_PATH_PATTERN = Pattern.compile(
+        String.format(PATH_PARAM_REGEX, TRANSCRIPTIONS_PATH));
+
+    private final Authorisation authorisation;
+
+    @Before("@annotation(uk.gov.hmcts.darts.authorisation.annotation.Authorisation)")
+    public void authorisation(JoinPoint joinPoint) throws Throwable {
+        HttpServletRequest request = ((ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()).getRequest();
+
+        uk.gov.hmcts.darts.authorisation.annotation.Authorisation authorisationAnnotation = ((MethodSignature) joinPoint.getSignature()).getMethod()
+            .getAnnotation(uk.gov.hmcts.darts.authorisation.annotation.Authorisation.class);
+
+        Set<SecurityRoleEnum> roles = Set.of(authorisationAnnotation.securityRoles());
+        if (roles.isEmpty()) {
+            throw new DartsApiException(AuthorisationError.USER_NOT_AUTHORISED_FOR_COURTHOUSE);
+        }
+
+        ContextIdEnum contextId = authorisationAnnotation.contextId();
+
+        switch (contextId) {
+            case CASE_ID -> checkAuthorisationByCaseId(request, roles);
+            case HEARING_ID -> checkAuthorisationByHearingId(request, roles);
+            case MEDIA_REQUEST_ID -> checkAuthorisationByMediaRequestId(request, roles);
+            case MEDIA_ID -> checkAuthorisationByMediaId(request, roles);
+            case TRANSCRIPTION_ID -> checkAuthorisationByTranscriptionId(request, roles);
+            default -> log.warn("Unrecognised contextId");
```

Should this throw an exception when the context is unknown?

---

It should never happen, I only added the default line because of checkstyle `switch without "default" clause. [MissingSwitchDefault]`. Done in commit ab8c4036.